lib/vnet/daemon/client_darwin.h

```diff
@@ -0,0 +1,53 @@
+#ifndef client_darwin_h
+#define client_darwin_h
+
+#import <Foundation/Foundation.h>
+
+typedef struct BundlePathResult {
+    const char * bundlePath;
+} BundlePathResult;
+
+// BundlePath updates bundlePath of result with the path
+// to the bundle directory that contains the current executable.
+// bundlePath is an empty string if the bundle details could not be fetched.
+// It might return a path even for executables that are not in a bundle.
+// In that case, calling codesign --verify on that path will simply return with 1.
+void BundlePath(struct BundlePathResult *result);
+
+// Returns the label for the daemon by getting the identifier of the bundle
+// this executable is shipped in and appending ".vnetd" to it.
+//
+// The returned string might be empty if the executable is not in a bundle.
+//
+// The filename and the value of the Label key in the plist file and the Mach
+// service of of the daemon must match the string returned from this function.
+NSString * DaemonLabel(void);
+
+// DaemonPlist takes the result of DaemonLabel and appends ".plist" to it
+// if not empty.
+NSString * DaemonPlist(void);
+
+typedef struct RegisterDaemonResult {
+    bool ok;
+    // service_status is fetched even if [service registerAndReturnError] fails,
+    // for debugging purposes.
+    int service_status;
```

Comment:
nit: Lower camel (bundlePath) or lower_snake (service_status)?  :P

Reply:
I'm going to stick to lower_snake for fields. `bundlePath` was one that I forgot to update. ;)